### PR TITLE
Keep language code in filename

### PR DESCRIPTION
**Pull request by erikh360**
Welcome messages are in different folders based on their language.

### CODE REVIEW

File: message_sender/factory.py [hunk 1]
```diff
@@ -47,6 +48,21 @@ def __init__(self, url, auth=None, from_addr=None, session=None,
         self.override_payload = override_payload
         self.strip_filepath = strip_filepath
 
+    def _get_filename(self, path):
+        """
+        This function gets the base filename from the path, if a language code
+        is present the filename will start from there.
+        """
+        match = re.search('[a-z]{2,3}_[A-Z]{2}', path)
```
Comment (OMosimege) [inline message_sender/factory.py:56]: Hi! Code is looking great cake I just have a few questions, as part of my learning process:

Comment (OMosimege) [reply message_sender/factory.py:56]: I see here that we are trying to search some pattern "path", but what are we trying to do with the '[a-z]{2,3}_[A-Z]{2}'

File: message_sender/tests.py [hunk 2]
```diff
@@ -2017,6 +2017,51 @@ def test_send_voice_override_payload_multiple_urls(self):
         self.assertEqual(r['nested_data']['unknown'], 'unknown')
         self.assertEqual(r['filename'], ['test1.mp3', 'test2.mp3'])
 
+    @responses.activate
+    def test_send_voice_strip_fielpath_language(self):
```
Comment (OMosimege) [inline message_sender/tests.py]: the function name, did you maybe mean "filepath" instead of "fielpath"?

File: message_sender/factory.py (hunk 1 above)
Comment (dubesoftware) [reply message_sender/factory.py:56]: @OMosimege the '[a-z]{2,3}_[A-Z]{2}' is a regular expression to match a language code in the file path. It's saying, find a substring matching the given pattern, e.g. 'en_US', 'en_GB' etc and return a match object (or None if the match is not found).

https://docs.python.org/3/library/re.html

For more on regex, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions

File: message_sender/tests.py (hunk 2 above)
Comment (erikh360) [reply message_sender/tests.py]: Haha thanks, good catch!

**Comment by erikh360** [comment]
@OMosimege Ready for another look, thanks!

File: message_sender/factory.py (hunk 1 above)
Comment (OMosimege) [reply message_sender/factory.py:56]: Thanks @dubesoftware

File: message_sender/tests.py [hunk 3]
```diff
@@ -2018,7 +2018,7 @@ def test_send_voice_override_payload_multiple_urls(self):
         self.assertEqual(r['filename'], ['test1.mp3', 'test2.mp3'])
 
     @responses.activate
-    def test_send_voice_strip_fielpath_language(self):
+    def test_send_voice_strip_filepath_language(self):
```
Comment (OMosimege) [inline message_sender/tests.py]: +1

**Review by OMosimege — APPROVED** [review]

**Pull request merged**